### ISSUE TITLE
Weird results on Travis for tests that work locally

### ISSUE THREAD

**Issue by mizzao** [arunoda/travis-ci-meteor-packages#36]
I most recently ran tests with Meteor 1.0.2.1 and they are timing out on Travis with very weird results, see https://travis-ci.org/HarvardEconCS/turkserver-meteor/builds/45494318

The package is https://github.com/HarvardEconCS/turkserver-meteor and tests work normally on local machine, both in browser and with `test-in-console` - you can verify that too. However on travis there appears to be some strange client/server connection errors and the test times out after 50 minutes.

Any idea what might be happening here? Previous tests (basically identical) seem to have worked with node v0.10.33 which implies Meteor 1.0.2: https://travis-ci.org/HarvardEconCS/turkserver-meteor/builds/43033496

**Comment by mizzao** [comment]
Is there a way to simulate the way that PhantomJS runs the tests locally, without having to push to Travis to test? It's somehow different from both `test-in-console` and `test-in-browser`.

**Comment by mitar** [comment]
Yes. :-)

**Comment by fix** [comment]
As you mentioned @mizzao i have the same problem and when i run the test locally using `make test`. it get stuck (ie no output, nothing happens). What about you?

Maybe the test loop to exit test is wrong and for some reason spawning new tests ?

**Comment by mitar** [comment]
Maybe this is related to [this change](https://github.com/meteor/meteor/pull/2357#issuecomment-72785545)? Maybe this package should now set `METEOR_PRINT_ON_LISTEN` and then wait for `LISTENING`?

**Comment by fix** [comment]
any idea @arunoda ?

**Comment by mizzao** [comment]
My test doesn't get "stuck" in the sense that the client keeps reconnecting and spawns new tests each time. However this only seems to affect some packages - others work fine.

It's not apparently clear how meteor/meteor#2357 affects it, but as I [mentioned here](https://groups.google.com/forum/#!topic/meteor-core/_JeiDvtYBFU) things broke sometime between 1.0.2 and 1.0.3.1 so it could be related; I'll look into it.

**Comment by fix** [comment]
Hi @mizzao any progress on your side on this issue ?

**Comment by mizzao** [comment]
No, but @spitis has discovered that the tests work differently in Linux vs OS X, possibly due to an issue in PhantomJS (the Meteor version change may have been a red herring). Perhaps he will comment in this thread.

**Comment by mizzao** [comment]
I've narrowed it down to an apparent change in the Travis CI environment, not the Meteor environment. Travis CI is now using PhantomJS 1.9.8; not sure what it was doing before but something different.

**Comment by mizzao** [comment]
I'm not sure why, but the hiccups go away running on OS X instead of Linux:

https://travis-ci.org/HarvardEconCS/turkserver-meteor/builds/55751466

using this workaround: https://github.com/HarvardEconCS/turkserver-meteor/commit/a95994affdcff64b69a0b8eec03330c23cfa9c5d

Really not sure what is going on here, so I'm going to punt the problem (in PhantomJS, Node, whatever) off to someone else to fix for now :)

**Comment by fix** [comment]
Well done @mizzao, you made my day, the workaround fixed the build!

You deserve a beer!
Thanks!

**Comment by mizzao** [comment]
There are a lot of things that changed here, including probably the PhantomJS version, the NodeJS version, and the OS. Your tests are a lot simpler than mine and would probably help to narrow down exactly what the issue is (both seem to involve packages that try to test client-to-server functions).

Maybe if someone else has an issue they will also find this thread :) My best guess right now is that the Linux PhantomJS is not maintaining a WebSocket connection properly to the Meteor server.

**Comment by mitar** [comment]
All my tests still run fine on Travis CI. The only recent change is the sudo requirement switch.

**Comment by mizzao** [comment]
@mitar all my tests run fine except for the package that does some weird client-server stuff in Tinytest :)

**Comment by mitar** [comment]
Oh, you need a newer Phantom? Yes, 2.0 solves many issues.

**Comment by mizzao** [comment]
I dunno if it was the newer Phantom that did it. That was just the newest version I could download for the build.

I think it was just switching to OS X that did it.

**Comment by fix** [comment]
yes, I can confirm, switching to OS X did the trick
